iothub/device/src/InternalClient.cs

```diff
@@ -1541,7 +1682,7 @@ public async Task SetInputMessageHandlerAsync(string inputName, MessageHandler m
             try
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                await _receiveSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
+                await _moduleReceiveMessageSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
renamed this to differentiate between the receiving semaphore for devices (newly added) vs the one for modules (pre-existing).

Reply:
Is the underlying implementation different for both?